frontend/javascripts/oxalis/model_initialization.ts

```diff
@@ -785,6 +790,28 @@ function enforcePricingRestrictionsOnUserConfiguration(
   return userConfiguration;
 }
 
+function ensureDatasetSettingsHasLayerOrder(
+  datasetConfiguration: DatasetConfiguration,
+  dataset: APIDataset,
+): DatasetConfiguration {
+  const colorLayerNames = _.keys(datasetConfiguration.layers).filter((layerName) =>
+    isColorLayer(dataset, layerName),
+  );
+  if (
+    datasetConfiguration.colorLayerOrder == null ||
+    datasetConfiguration.colorLayerOrder.length < colorLayerNames.length
+  ) {
+    return {
+      ...datasetConfiguration,
+      colorLayerOrder: colorLayerNames,
+    };
+  }
+  const onlyExistingLayers = datasetConfiguration.colorLayerOrder.filter(
+    (layerName) => colorLayerNames.indexOf(layerName) >= 0,
+  );
```

Comment:
this guards against the case in which `colorLayerOrder` contains additional names that don't exist in the dataset, right? however, if this happens, couldn't it be that `onlyExistingLayers.length < colorLayerNames.length`? How about moving this filtering over the if-block in line 800?